.ci/eslint-plugin-zotero-translator/index.js

```diff
@@ -6,68 +6,93 @@
 
 const fs = require('fs');
 const path = require('path');
+const child_process = require('child_process');
+
+const repo = path.resolve(path.join(__dirname, '..', '..'));
+function exec(cmd, cwd=null) {
+	return child_process.execSync(cmd, { cwd: cwd || repo, encoding: 'utf8' });
+}
 
 const header_var_name = '__eslint_zotero_translator_header';
 
 const headers = {};
 const deleted = new Set(
-	fs.readFileSync('deleted.txt', 'utf-8')
+	fs.readFileSync(path.join(repo, 'deleted.txt'), 'utf-8')
 		.split('\n')
 		.map(line => line.split(' ')[0])
 		.filter(id => id && id.indexOf('-') > 0)
 );
 
+const branch = exec('git rev-parse --abbrev-ref HEAD').trim();
+// branch to compare lastUpdated against -- assume that if you're not in CI, you have upstream/master
+const master = process.env.CI === 'true' ? 'master' : 'upstream/master';
+
+const lastUpdated = {};
+if (branch !== master) {
+	const updated = exec(`git diff --name-status ${master}`)
+		.split('\n')
+		.map(changed => changed.match(/^M\t([^\/]+\.js)$/))
+		.filter(changed => changed)
+		.map(changed => changed[1]);
+
+	for (const lu of child_process.execSync(`git grep '"lastUpdated"' ${master} *.js`, { cwd: repo, encoding: 'utf8' }).split('\n')) {
+		const m = lu.match(/^[a-z\/]+:([^:]+):\s*"lastUpdated"\s*:\s*"([-0-9: ]+)"/);
+		if (m && updated.includes(m[1])) lastUpdated[m[1]] = m[2];
+	}
+}
+
```

Comment:
Can you comment this a bit so it's clearer what's happening?

Reply:
Ah, I see you followed up with an explanation. Would be good to just add most of that to a comment so it's available in the code.

Reply:
I've added it as a comment above that block.